app/models/parameter.rb

```diff
@@ -10,6 +10,29 @@ class Parameter < ActiveRecord::Base
   validates :name, :presence => true, :no_whitespace => true
 
   scoped_search :on => :name, :complete_value => true
+  scoped_search :on => :type, :complete_value => true
+  scoped_search :on => :value, :complete_value => true
+
+  # children associations must be defined here, otherwise scoped search definitions won't find them
+  belongs_to :domain, :foreign_key => :reference_id, :inverse_of => :domain_parameters
```

Comment:
Yes, I see your point, thanks for trying. It seems possible to re-open the Parameter class before declaring DomainParameter, e.g.

```ruby
require_dependency 'parameter'
class Parameter
  # associations must be defined on superclass, otherwise scoped search definitions does not find them
  belongs_to :domain, :foreign_key => :reference_id, :inverse_of => :domain_parameters
  # scoped search must be defined on superclass, so the same terms can be used for searching in parent class
  scoped_search :in => :domain, :on => :name, :complete_value => true, :rename => 'domain_name'
end

class DomainParameter < Parameter
[..]
end
```

but the complexity's high and I don't think it's worth it (as plugins may not subclass Parameter correctly in any case).